### DISCUSSION TITLE
Alore Programlama | SonsuzUs Github

### DISCUSSION THREAD

**Discussion by giscus** [sonsuzus/sonsuzus.github.io#358] [Site]
# Alore Programlama | SonsuzUs Github

Alore

https://sonsuzus.github.io/posts/alore-programlama/

**Comment by sonsuzus** [comment]
```alore
fun is_prime(n) {  
    if (n < 2) {  
        return false  
    }  

    for (i = 2; i * i <= n; i = i + 1) {  
        if (n % i == 0) {  
            return false  
        }  
    }  
    return true  
}  

fun main() {  
    for (n = 2; n <= 1000; n = n + 1) {  
        if (is_prime(n)) {  
            print(n + "\n")  
        }  
    }  
}  

main()
```

is_prime Fonksiyonu: Verilen sayının asal olup olmadığını belirler. Sayı 2’den küçükse false döner. 2’den başlayarak sayının kareköküne kadar olan bölenleri kontrol eder. Eğer herhangi biri tam bölünüyorsa, sayı asal değildir ve false döner.

main Fonksiyonu: 2’den 1000’e kadar olan sayılar üzerinde döngü yapar. Her sayıyı is_prime fonksiyonu ile kontrol eder ve eğer sayı asal ise yazdırır.
